cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -711,9 +719,62 @@ pub fn path_hash(p: &impl AsRef<Path>) -> String {
     chars.to_string()
 }
 
+/// Return a path to the services socket given a unique identifier
+///
+/// Socket paths cannot exceed 104 characters on macOS
+/// - TMPDIR will often have a long path, e.g.
+///   /var/folders/8q/spckhr654cv4xrcv0fxsrlvc0000gn/T/nix-shell.vfDA8u
+/// - /var/run is not writeable
+/// So we use /tmp
+///
+/// On Linux use XDG_RUNTIME_DIR per
+/// https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+/// If unset, fallback to /tmp
+fn services_socket_path(id: &str, flox: &Flox) -> Result<PathBuf, EnvironmentError> {
+    #[cfg(target_os = "macos")]
```

Comment:
Nitpick: I find it easier to read if the various OS options for a single piece of data are grouped together rather than doing all the data for one OS and the all the data for the other OS.

Reply:
Fixed